### ISSUE TITLE
"Reindent Lines" behaves differently for square brackets

### ISSUE THREAD

**Issue by BastiHz** [rstudio/rstudio#4260]
### System details

    RStudio Edition: Desktop
    RStudio Version: 1.1.463 and 1.2.1256
    OS Version: Windows 7 and Kubuntu 18.04
    R Version: 3.5.2

### Steps to reproduce the problem

1. Write the following code:
```
foo(
    bar,
    baz
)
foo {
    bar,
    baz
}
foo[
    bar,
    baz
]
```
2. Highlight everything.
3. Reindent lines (Ctrl+I)

### Describe the problem in detail

The indentation of the parentheses and braces remain unchanged but the closing square bracket becomes indented: 
```
foo(
    bar,
    baz
)
foo {
    bar,
    baz
}
foo[
    bar,
    baz
    ]        <- this one is different
```

### Describe the behavior you expected
In my opinion the indentation should be the same no matter what kind of brackets/braces/parentheses is used.

**Comment by ronblum** [comment]
@BastiHz Thanks for noting this. I checked several style guides and other resources, such as the [Tidyverse Style Guide](https://style.tidyverse.org/), and none of them address or have examples of square brackets over multiple lines. So, I'll mark it for now as an enhancement to be reviewed as we continue development of RStudio.

**Comment by jmcphers** [comment]
Fixed in https://github.com/rstudio/rstudio/pull/4309.

**Comment by dfalty** [comment]
Verified fixed 1.3.673
